cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -154,13 +180,27 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   private def readFrames(event: RecordedEvent): List[CallSite] = {
     val stack = event.getStackTrace
-
+    val isNative = event.getEventType.getLabel=="Method Profiling Sample Native"
     if (stack eq null) List() else {
       stack.getFrames.iterator.asScala.map {
         frame: RecordedFrame =>
           val method = frame.getMethod
-          CallSite(method.getType.getName, method.getName, method.getName, method.getDescriptor, frame.getLineNumber)
-      }.distinct.takeWhile { f => !f.isIgnorableTopFrame }.toList
+          if(method ne null) {
+            val splitIndex = method.getType.getName.lastIndexOf(".");
+            CallSite(
+              method.getType.getName.substring(0,splitIndex).intern(),
+              method.getType.getName.substring(splitIndex+1).intern(),
+              method.getName.intern(),
+              isNative,
```

Comment:
isNative is part of the event, not the CallSite
We need to record information based on the sample being native or not